examples/knative/go/kn-go-tagging/README.md

```diff
@@ -0,0 +1,99 @@
+
+`ko` was created to help Knative developers build images and binaries without the need for Dockerfiles nor Makefiles.
+See more information in [this Knative blog](https://knative.dev/blog/articles/ko-fast-kubernetes-microservice-development-in-go/).
+
+Follow setup instructions at https://github.com/google/ko.
+
+You will also need to install [Go](https://golang.org/doc/install) for the ko tool to work.
+
+After installing ko, set the destination for images with an environment variable.
+
+```bash
+KO_DOCKER_REPO=my-dockerhub-user
+```
+
+Run the following command from the `kn-go-echo` directory. It will build and push the image to your local Docker daemon. To have `ko` publish and use a container image registry, remove the `--local` flag.
+
+```bash
+ko publish --local .
+```
+
+Save the image name so you can use it in the [config/knative.yaml]'s Service container image section (mentioned in Step 3).
+
+## Step 2: Create vSphere Credentials Secret
+
+Create a secret holding the username and password needed to access vCenter.
+
+```bash
+kubectl create secret generic vsphere-credentials \
+--from-literal=username=administrator@vsphere.local \
+--from-literal=password='Admin!23' \
+--namespace vmware-functions
+```
+
+## Step 3: Configure the Knative Service and Trigger
+
+The [config/knative.yaml] will need to be updated to run the Knative Service.
+
+Get the name of the tag that will be attached to the VM by creating a tag within vSphere.
+Put the tag name in the `VALUE` field of `TAG_Name` for the Service manifest.
+
+You can specify if the tag should be attached or detached to the VM that triggers the event by setting `TAG_ACTION` to `"attach"` or `"detach"`.
+
+These are the configuration values that can be set in the Knative Service spec.template.spec.containers portion of the [Knative manifest](config/knative.yaml):
+
+| Configuration       | Description                                                   | Example Values             | Required/Optional |
+| ------------------- | ------------------------------------------------------------- | -------------------------- | ----------------- |
+| image               | image name generated by ko                                    | ko.local/kn-go-tagging:0.1 | Required          |
+| VCENTER_INSECURE    | Use TLS to connect to vCenter or not                          | "true" or "false"          | Optional          |
+| VCENTER_SECRET_PATH | The path where the vSphere credentials secret will be mounted | "/var/bindings/vsphere"    | Optional          |
+| DEBUG               | Set the logging verbosity for help with service development   | "true" or "false"          | Optional          |
+| TAG_NAME            | Name given to a tag created in vSphere                        | "some-name"                | Required          |
+| TAG_ACTION          | Action for tag when specified VM event reaches service        | "attach" or "detach"       | Optional          |
```

Comment:
nit: this field is also a `required` field as per your envConfig

Reply:
I'll make this default "attach" and not required in the envConfig.

Also, I'll validate to ensure the only allowed options are "attach" and "detach" if it's not already there.

Reply:
nit: can you please leave a note what the default is? unclear from the table (I know it's `attach` bc you told me)